apps/cbportal/base/cloudbroker__grid/methodclass/cloudbroker_grid.py

```diff
@@ -93,6 +93,14 @@ def runUpgradeScript(self, **kwargs):
                                             {'$set': {'status': 'CURRENT', 'updateTime': j.base.time.getTimeEpoch() }})
         return 'Upgrade script ran successfully'
 
+
+    @auth(groups=['level1', 'level2', 'level3'], skipversioncheck=True)
+    def upgradeFailed(self, **kwargs):
+        if not self.sysmodels.version.searchOne({'status': 'INSTALLING'}):
+            return False
+        self.sysmodels.version.updateSearch({'status': 'INSTALLING'}, {'$set': {'status': 'ERROR'}})
```

Comment:
you can just make this call and based on the nModified know if something was changed and return based on this value